src/language-js/comments.js

```diff
@@ -884,6 +900,48 @@ function handleTSMappedTypeComments(
   return false;
 }
 
+function handleLogicalExpression(
+  text,
+  enclosingNode,
+  precedingNode,
+  followingNode,
+  comment,
+  options
+) {
+  if (enclosingNode && enclosingNode.type === "LogicalExpression") {
+    if (
+      followingNode &&
+      isBlockComment(comment) &&
+      privateUtil.hasNewline(text, options.locEnd(comment))
+    ) {
+      addLeadingComment(followingNode, comment);
+      return true;
+    }
+  }
+  return false;
+}
+
+function handleBinaryExpression(
+  text,
```

Comment:
`handleLogicalExpression` and `handleBinaryExpression` are almost identical. Please make them one function.

Reply:
@thorn0 OK, I fixed it on https://github.com/prettier/prettier/pull/7787/commits/51f1b50df4ce5b5f2da8211aad77c3d210628fe8
Though I named the function `handleEndOfLineBlockComments`, I am worrying if it is suitable for it. Would you give me some advice if you have any idea?

Reply:
The binary expression name was good. Logical expressions are binary too.

Reply:
Thanks. I fixed on https://github.com/prettier/prettier/pull/7787/commits/3f743fd26274e3cddd97aa223c5e5524d5dddd38.